lib/honeybadger/cli/check_ins.rb

```diff
@@ -0,0 +1,55 @@
+require 'forwardable'
+require 'honeybadger/cli/main'
+require 'honeybadger/cli/helpers'
+require 'honeybadger/util/http'
+require 'honeybadger/config_sync_service'
+module Honeybadger
+  module CLI
+    class CheckIns < Thor
+
+      extend Forwardable
+
+      class_option :personal_auth_token, required: false, type: :string, desc: "personal auth token for API access"
+      class_option :environment,     required: false, aliases: [:'-e', :'-env'], type: :string, desc: 'Environment this command is being executed in (i.e. "production", "staging")'
```

Comment:
What would be the use case to call the sync CLI command per environment?

I understand that when executing the check-in api call, we could allow passing the environment variable as a parameter in order to dictate that the check-in api call should be made only when the environment matches (example [here](https://docs.honeybadger.io/lib/php/integration/laravel/#after-a-scheduled-command)).

However, this is the sync CLI command and ideally it should only run on the production environment, right?

Reply:
@subzero10 The others may prove me wrong, but I think this could be important as rails technically does allow for multiple "production" environments, as it is not restricted to "development", "test" and "production" as other frameworks. So if the honeybadger config is actually set in a rails initializer, it could be environment dependent.

So, for example, the sync cli command could also run for a "staging" environment" and that could load/should load different checks.

Reply:
> So, for example, the sync cli command could also run for a "staging" environment" and that could load/should load different checks.

I see, but I think that wouldn't work in this case. As far as I know, the backend does not support check-ins per environment.
However, I'm wondering how would you know which config file to read if no environment is passed as an argument. Or in other words, which one is the "production" environment?

Reply:
Maybe @stympy or @joshuap could chime in here. My thinking was that a honeybadger initializer in a rails app could do checks like `if Rails.env.staging?` and do different things depending on the loaded environment. I would assume different "production" environments would use different Honeybadger projects altogether and thus also would use different checkins. 

This might all be a bit theoretical but the fact that config *can* be set from within Rails configuration files/initializers makes me quite convinced the environment is important information the user should be able to pass in here.

Reply:
> I would assume different "production" environments would use different Honeybadger projects altogether and thus also would use different checkins.

That is a fair point that I didn't consider 😄. This isn't a blocker for me. We can proceed as it is and we may come back to it in the future if it causes confusion.

Reply:
I see you can set the `project_id` per check-in in the `check_ins` config, so maybe that's enough—people can sync check-ins for one or many projects already?

Reply:
> I think the best course here is to build the simplest thing (don't consider environments at all in the Rails integration), and then we could tackle an official solution later if people want it. In the meantime, it would be nice if we could suggest a manual workaround for anyone who does want to use multiple Honeybadger projects in the same Rails app.

I agree with this approach and that's what prompted me to ask about the _environment_ option in the first place 😄.

> I see you can set the project_id per check-in in the check_ins config, so maybe that's enough—people can sync check-ins for one or many projects already?

Indeed, that is a consequence of having to supply a `project_id`, since we can't use the project API key at the moment.